nemo/collections/asr/parts/utils/transcribe_spk_utils.py

```diff
@@ -0,0 +1,388 @@
+        raise ValueError("Either filepaths or manifest shoud not be None")
+
+    if manifest:
+        filepaths = []
+        offsets = []
+        durations = []
+        query_audio_files = []
+        query_offsets = []
+        query_durations = []
+
+        #separater info
+        separater_freq = asr.asr_model.cfg.test_ds.separater_freq
+        separater_duration = asr.asr_model.cfg.test_ds.separater_duration
+        separater_unvoice_ratio = asr.asr_model.cfg.test_ds.separater_unvoice_ratio
+        print(f'Batchview: dataset')
+        with open(manifest, "r", encoding='utf_8') as mfst_f:
+            print("Parsing manifest files...")
+            for l in mfst_f:
+                row = json.loads(l.strip())
+                audio_file = get_full_path(audio_file=row['audio_filepath'], manifest_file=manifest)
+                offset = row['offset']
+                duration = row['duration']
+                #query info
+                query_audio_file = row['query_audio_filepath']
+                query_offset = row['query_offset']
+                query_duration = row['query_duration']
+                #save to list, each element corresponding to attribute of one sample
+                offsets.append(offset)
+                durations.append(duration)
+                query_audio_files.append(query_audio_file)
+                query_offsets.append(query_offset)
+                query_durations.append(query_duration)
+                
+                filepaths.append(audio_file)
+                if 'text' in row:
+                    refs.append(row['text'])
+        assert len(refs) == len(filepaths) == len(offsets) == len(durations) == len(query_audio_files) == len(query_offsets) == len(query_durations)
+
+    with torch.inference_mode():
+        with torch.amp.autocast('cpu' if accelerator == 'cpu' else 'cuda'):
+            batch = []
+            asr.sample_offset = 0
+            for idx in tqdm(range(len(filepaths)), desc='Sample:', total=len(filepaths)):
+                batch.append((filepaths[idx], offsets[idx], durations[idx], query_audio_files[idx], query_offsets[idx], query_durations[idx])
+                )
+            
+                if len(batch) == batch_size:
+                    batch_audio_files = [sample[0] for sample in batch]
+                    batch_offsets = [sample[1] for sample in batch]
+                    batch_durations = [sample[2] for sample in batch]
+                    batch_query_audio_files = [sample[3] for sample in batch]
+                    batch_query_offsets = [sample[4] for sample in batch]
+                    batch_query_durations= [sample[5] for sample in batch]
+
+                    asr.reset()
+                    asr.read_audio_file(batch_audio_files, batch_offsets, batch_durations, batch_query_audio_files, batch_query_offsets, batch_query_durations, separater_freq, separater_duration, separater_unvoice_ratio, delay, model_stride_in_secs)
```

Comment:
Again, there should be variable assignment out side if statement (which guarantees always existing variables)

e.g., batch_audio_files = [], ...